### ISSUE TITLE
print error status of execv() and /usr/sbin/pppd

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [adrienverge/openfortivpn#201]
It looks like `fprintf(stderr, ...)` doesn't print in this piece of code of _tunnel.c_:
```
		execv(args[0], args);
		fprintf(stderr, "execv: %s\n", strerror(errno));
		exit(EXIT_FAILURE);
```

See issue #194 where no error message is printed despite the missing `/usr/sbin/pppd` executable.

An additional error message has been added in pull request #199 before a `forkpty()` call, with the rationale that the `forkpty()` might be responsible for the `fprintf(stderr, ...)` failure.

Now if we find a solution to print an error message just after the `execv()` call, we might revert #199.

For the record, this piece of code originates in pull request #152  where I had added a [comment](https://github.com/adrienverge/openfortivpn/pull/152#issuecomment-318321587) suggesting an alternative implementation:
> An alternative would be to rewrite _do_log()_ as follows and use the mutex-less _unprotected_do_log()_ after _execvp()_:
> ```
> void do_log(int verbosity, const char *format, ...)
> {
> 	va_list args;
> 
> 	pthread_mutex_lock(&mutex);
> 
> 	unprotected_do_log(verbosity, format, args);
> 
> 	pthread_mutex_unlock(&mutex);
> }
> ```
I doubt this alternative implementation helps - as I see it the real problem is the `forkpty()` call. It's nevertheless a good idea to implement it with a different goal: consistency in error output.

**Comment by DimitriPapadopoulos** [comment]
This is clearly an effect of `forkpty()`.

The child process operates in a pseudoterminal. If there's an error in the child process - for example when executing a program with execv() - I cannot find a way to access / print to the initial pre-fork stderr.

**Comment by DimitriPapadopoulos** [comment]
#### Extended goals

Make sure error messages are logged and printed to `stderr` in both these cases:
1. `execv()` fails to execute `/usr/sbin/pppd` (for example because PPPD is not installed on the system),
2. `/usr/sbin/pppd` is executed but fails (for example because option `ifname` is not implemented in the available version of PPPD): we already retrieve the exit status code and print error messages (#189) but these messages are not always clear, so it would be better to intercept the `stderr` output of `/usr/sbin/pppd` in the parent process and reprint (perhaps after filtering) to the parent process' `stderr`.

#### How to achieve that?

See #210.

**Comment by DimitriPapadopoulos** [comment]
Additionally I suspect that `stdout` and `stderr` from child process `/usr/sbin/pppd` are mixed into a single stream in the parent process because of _forkpty_(). Perhaps this could explain some of the "I/O errors" that have been reported...

**Comment by DimitriPapadopoulos** [comment]
I think this has been fixed by ae47371.
